torch_xla/csrc/ir.h

```diff
@@ -136,6 +136,16 @@ class XlaNode : public torch::lazy::Node {
 
   std::string ToString() const override;
 
+  std::vector<torch::lazy::NodePtr>* operands_as_nodes() { return &operands_; }
+  torch::lazy::Node* get_operand(size_t index) {
+    return operands_[index].get();
+  }
+  void set_operand(torch::lazy::NodePtr node, size_t index) {
```

Comment:
Hmm, why do we need to do it? If we change the operand, we might need to update the hash which will affect all nodes that take current node as input.

Reply:
So the output of update may not be a straight data node, but data node attached to another node. The data node will be an operand. When we set sharding, we will be creating a new node with a new hash. I haven't thought about updating the hash of the op holding the new data nodes as operlands.

Reply:
It looks like operands are not part of hash, and each node computes hash independently https://github.com/pytorch/pytorch/blob/b818b3fe1c1fa90529b8500cd8ef800bec8415e8/torch/csrc/lazy/core/ir.cpp#L43

Reply:
I think operand is part of the hash, check 
https://github.com/pytorch/xla/blob/5b5dd27f20ec80706e2d88f59b1826a5924c66f2/torch_xla/csrc/ir.cpp#L45-L51
and
https://github.com/pytorch/xla/blob/5b5dd27f20ec80706e2d88f59b1826a5924c66f2/torch_xla/csrc/ir.h#L113-L118